### ISSUE TITLE
Due date minor issue

### ISSUE THREAD

**Issue by frenya** [RobTrew/tree-tools#9]
Hi,

first of all, thanks for the excellent script. Saved me a lot of work migrating tasks from my previous task manager!

I ran into a minor bug yesterday. When setting the due date, you start by setting it to current date and then setting to components from the @due flag.

This way, there are a couple combinations of current date and due date that try to create invalid dates (e.g. Febuary 30).

It would be better to initialize the dueDate with 1/1/1900.

Regards,

Frenya

**Comment by RobTrew** [comment]
Frenya – thanks !  Very good point. I must look out for that in my scripts generally.

In the meanwhile, could you tell me which script in particular you found this in ?

All best,  Rob

**Comment by RobTrew** [comment]
PS One issue is that setting Applescript dates from strings that work in one locale may cause errors in another, but a workaround will be to start with a current date and then reset its value numerically, but more radically.

**Comment by frenya** [comment]
Hi,

sorry, I didn't mention that. I was using the main.scpt script to import
TaskPaper formatted text into OmniFocus. And yeah, I figured there's
probably a reason for starting with the current date.

I used a quick and dirty fix - just reversed the order of setting the day
and the month - which works fine in January, but would fail in February :)

Setting the day to 1, then the month and then the day would be an ugly but
effective solution ;)

Take care,

Frank
